src/main/java/com/klodnicki/Bike/v2/model/entity/Rent.java

```diff
@@ -0,0 +1,43 @@
+package com.klodnicki.Bike.v2.model.entity;
+
+import jakarta.persistence.*;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.time.LocalDateTime;
+
+@Entity
+@Getter
+@Setter
+@NoArgsConstructor
+public class Rent {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", nullable = false)
+    private Long id;
+    private LocalDateTime rentalStartTime;
+    private LocalDateTime rentalEndTime;
+    private int daysOfRent;
+    @OneToOne
+    @JoinColumn(name = "bike_id")
+    private Bike bike;
+
+    @OneToOne
+    @JoinColumn(name = "user_id")
+    private User user;
+
+    @OneToOne
+    @JoinColumn(name = "chargingStation_id")
+    private ChargingStation chargingStation;
+
+    public Rent(LocalDateTime rentalStartTime, LocalDateTime rentalEndTime, Bike bike, User user,
+                ChargingStation chargingStation, int daysOfRent) {
+        this.rentalStartTime = rentalStartTime;
+        this.rentalEndTime = rentalEndTime;
+        this.bike = bike;
+        this.user = user;
+        this.chargingStation = chargingStation;
+        this.daysOfRent = daysOfRent;
+    }
```

Comment:
Lombok could be used here

Reply:
I cannot use lombok annotation for constructor here, because not all fields are to be in constructor. On the other hand I cannot use @RequiredArgsConstructor because it demands to have a final field and I dont want to have Long id field as final. That is why this constructor is created manually.

Reply:
But there is every field here as far as I am concerned. I believe it's only missing `id` field which I don't see the reason why it couldn't be added to this constructor.

Reply:
Because when I put Long id field to the constructor then I must manually input the id here (which is not acceptable, because I want that id number will be incremented automatically)
![image](https://github.com/BartoliniAndBorderCollies/Bike.v2_REST_API/assets/126821059/796f1bc9-f927-4d69-8a52-0c7338975c8a)

Reply:
> Because when I put Long id field to the constructor then I must manually input the id here (which is not acceptable, because I want that id number will be incremented automatically)

You can do that anyway by having the id in the constructor and then in the line you outlined in red pass "null" as an id.
This will have the exact same outcome, as if you will check by debugger by using the current constructor without id, you still have id being null :)

Reply:
oh I didn't know that! that would be great! I'm going to check that right now!

Reply:
it works just fine! thanks for that!